server/src/main/java/io/crate/metadata/doc/DocIndexMetadata.java

```diff
@@ -435,8 +435,9 @@ private void internalExtractColumnDefinitions(@Nullable ColumnIdent columnIdent,
 
             boolean nullable = !notNullColumns.contains(newIdent) && !primaryKey.contains(newIdent);
             columnProperties = furtherColumnProperties(columnProperties);
-            assert columnProperties.containsKey("position") : "Column position is missing: " + newIdent.fqn();
-            int position = (int) columnProperties.get("position");
+            assert columnProperties.containsKey("position") && columnProperties.get("position") != null : "Column position is missing: " + newIdent.fqn();
+            // BWC compatibility with nodes < 5.1, position could be NULL if column is created on that nodes
+            int position = (int) columnProperties.getOrDefault("position", 0);
```

Comment:
wouldn't an assert happen before we get to this code?

Reply:
assertions aren't enabled in production.

Reply:
ah, ok, put back ready-to-merge label.